concent_api/core/validation.py

```diff
@@ -116,43 +100,33 @@ def validate_task_to_compute(task_to_compute: message.TaskToCompute):
         'provider_public_key',
         'requestor_public_key'
     ]])):
-        raise Http400(
+        raise ConcentValidationError(
             "Invalid TaskToCompute",
             error_code=ErrorCode.MESSAGE_WRONG_FIELDS,
         )
 
-    validate_int_value(task_to_compute.compute_task_def['deadline'])
+    validate_value_is_int_convertible_and_positive(task_to_compute.compute_task_def['deadline'])
 
     validate_id_value(task_to_compute.compute_task_def['task_id'], 'task_id')
     validate_id_value(task_to_compute.compute_task_def['subtask_id'], 'subtask_id')
 
     validate_hex_public_key(task_to_compute.provider_public_key, 'provider_public_key')
     validate_hex_public_key(task_to_compute.requestor_public_key, 'requestor_public_key')
     validate_secure_hash_algorithm(task_to_compute.package_hash)
-    validate_subtask_price_task_to_compute(task_to_compute)
+    validate_positive_integer_value(task_to_compute.price)
     validate_frames(task_to_compute.compute_task_def['extra_data']['frames'])
 
 
 def validate_report_computed_task_time_window(report_computed_task):
```

Comment:
This whole check is part of logic view, and used only there, it should be moved there (the inside of the function), and raise directly `http400` in view.

Reply:
Well, quite frankly it is. But on the other hand, it is good that it was extracted to a function. Maybe we need a separate module for business logic validation and have all such functions stored there?